pennylane/data/data_manager/__init__.py

```diff
@@ -0,0 +1,314 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains functions for querying available datasets and downloading
+them.
+"""
+
+import typing
+from concurrent.futures import FIRST_EXCEPTION, ThreadPoolExecutor, wait
+from functools import lru_cache
+from pathlib import Path
+from time import sleep
+from typing import List, Optional, Union
+
+import requests
+
+from pennylane.data.base import Dataset
+from pennylane.data.base.hdf5 import open_hdf5_s3
+
+from .foldermap import DataPath, FolderMapView, ParamArg
+from .params import DEFAULT, FULL, format_params
+
+S3_URL = "https://datasets.cloud.pennylane.ai/datasets/h5"
+FOLDERMAP_URL = f"{S3_URL}/foldermap.json"
+DATA_STRUCT_URL = f"{S3_URL}/data_struct.json"
+
+
+@lru_cache(maxsize=1)
+def _get_foldermap():
+    """Fetch the foldermap from S3."""
+    response = requests.get(FOLDERMAP_URL, timeout=5.0)
+    response.raise_for_status()
+
+    return FolderMapView(response.json())
+
+
+@lru_cache(maxsize=1)
+def _get_data_struct():
+    """Fetch the data struct from S3."""
+    response = requests.get(DATA_STRUCT_URL, timeout=5.0)
+    response.raise_for_status()
+
+    return response.json()
+
+
+def _download_partial(
+    s3_url: str, dest: Path, attributes: typing.Iterable[str], overwrite_attrs: bool
```

Comment:
It could be shortened to just use `overwrite`

```suggestion
    s3_url: str, dest: Path, attributes: typing.Iterable[str], overwrite: bool
```